tmo-monitor.py

```diff
@@ -134,79 +150,217 @@ def sha256url(self, val1, val2):
     return self.base64url_escape(self.sha256(val1, val2))
 
 
+class Configuration:
+  def __init__(self):
+    # Set default values
+    self.reboot_now = False
+    self.skip_reboot = False
+    self.login = dict([('username', 'admin'), ('password', '')])
+    self.ping = dict([('interface', ''), ('ping_host', 'google.com'), ('ping_count', 1), ('ping_interval', 10)])
+    self.connection = dict([('primary_band', ''), ('secondary_band', 'n41'), ('enbid', '')])
+    self.reboot = dict([('uptime', 90), ('ping', False), ('4G_band', False), ('5G_band', False), ('enbid', False)])
+    self.general = dict([('print_config', False)])
+  
+    # Command line arguments override defaults & .env file
+    self.read_environment()
+    args = self.parse_commandline()
+    self.parse_arguments(args)
+
+    if self.skip_reboot and self.reboot_now:
+      print('Incompatible options: --reboot and --skip-reboot\n', file=sys.stderr)
+      self.parser.print_help(sys.stderr)
+      sys.exit(2)
+    if self.skip_reboot:
+      for var in {'ping', '4G_band', '5G_band', 'enbid'}:
+        self.reboot[var] = False
+    # if not self.login['username']:
```

Comment:
Because the script is normally ran in an unintended manner for most use cases, we should check to see if the script is being invoked interactively or not and use the commented out block for non-interactive invocations of the script.

e.g. `if sys.stdin and sys.stdin.isatty():`

Reply:
Good suggestion, fixed.